src/backend_pgsql.c

```diff
@@ -260,8 +260,12 @@ backend_authenticate(const char *service, const char *user, const char *passwd,
 			rc = PAM_USER_UNKNOWN;
 		} else if (!PQgetisnull(res, 0, 0)) {
 			char *stored_pw = PQgetvalue(res, 0, 0);
-			if (!strcmp(stored_pw, (tmp = password_encrypt(options, user, passwd, stored_pw)))) rc = PAM_SUCCESS;
-			free (tmp);
+			if (options->pw_type == PW_FUNCTION) {
+				if (!strcmp(stored_pw, "t")) { rc = PAM_SUCCESS; }
+			} else {
```

Comment:
That looks suspicious. Care to explain how this works? If stored password equals to "t", return success?